website/pages/blog/optimize-bundle-size-with-swc-and-graphql-codegen.mdx

```diff
@@ -0,0 +1,191 @@
+import { useGetCartByIdQuery } from './graphql/generated-hooks'
+const { data } = useGetCartByIdQuery({ variables: { id: '1' } })
+
+// You can write your operations directly next to your component and get the correct types:
+import { graphql } from './graphql'
+const GetCartByIdDocument = graphql(/** GraphQL */ `
+  query GetCartById($id: ID!) {
+    id
+    items {
+      id
+      ...CartItem
+    }
+  }
+`)
+// Awesome! We have the correct types for this operation even when we specify it next to our component!
+```
+
+However, this approach can come at a cost - the bundle size of your pages can increase
+significantly, especially as the number of queries and mutation grows. In this post, we will show
+you what the problem is and how to optimize your bundle size with the new
+[`@graphql-codegen/client-preset-swc-plugin`](https://www.npmjs.com/package/@graphql-codegen/client-preset-swc-plugin)
+plugin for your applications supported by the SWC compiler such as projects using
+[Next.js](https://nextjs.org) or [Vite](https://vitejs.dev/).
+
+## The Generated Code When Using the `client-preset`
+
+When using the `client-preset` without any additional optimization, your site's bundle can quickly
+become bloated with unnecessary imports on every page/ component where you use GraphQL. This is
+because by default `client-preset`'s `graphql` function that enables a great TypeScript DX, has to
+have all the GraphQL operations available to it to parse the operation and have the correct types.
+
+The generated code output for the `graphql` function is a function that looks similar to the
+following code, where code-splitting and tree-shaking is not easily possible.
+
+```ts
+const documents = {
+  'fragment CartItems on CartItem {\n name\n quantity \n  }\n  ': types.CartItemsFragmentDoc,
+  'query GetCartById($id: ID!) {\n cart(id: $id) {\n id \n items { \n id \n ...CartItem\n }\n totalItems\n}\n  }\n':
+    types.GetCartByIdDocument
+}
+export function graphql(
+  source: 'fragment CartItems on CartItem {\n name\n quantity\n }\n '
+): (typeof documents)['\n fragment CartItem on CartItem {\n name\n quantity\n }\n ']
+export function graphql(
+  source: 'query GetCartById($id: ID!) {\n cart(id: $id) {\nid\nitems {\nid\n...CartItem\n}\ntotalItems\n }\n }\n'
+): (typeof documents)['\n query GetCartById($id: ID!) {\n cart(id: $id) {\n id\n items {\n id \n ...CartItem \n } \n totalItems \n } \n } n']
+export function graphql(source: string): unknown
+export function graphql(source: string) {
+  // Here we access the documents object, so the bundler cannot make assumption over what is
+  //  actually used at runtime
+  return (documents as any)[source] ?? {}
+}
+```
+
+This means that every time you use the `graphql` function, you are importing basically all used
+operations in your project, even if those are not being used. This can be a problem if you have a
+lot of documents (query, mutation, and fragments) in your project.
```

Comment:
```suggestion
Every time you use the `graphql` function, all the operations in the project are imported, regardless of wether they are used or not. This can be a problem if you have a lot of documents (query, mutation, and fragments) in your project.
```